File: backend/src/main/kotlin/com/example/backend/models/RetourProduit.kt
package com.example.backend.models

import jakarta.persistence.*
import org.hibernate.annotations.ColumnDefault
import java.time.Instant
import java.time.LocalDateTime

@Entity
@Table(name = "retour_produit")
class RetourProduit {
@Id
  @GeneratedValue(strategy = GenerationType.IDENTITY) // Use IDENTITY for auto-increment
  @Column(name = "id", nullable = false)
  var id: Int? = null

  @ManyToOne(fetch = FetchType.LAZY)
  @JoinColumn(name = "vente_id")
  var vente: com.example.backend.models.Vente? = null

  @ManyToOne(fetch = FetchType.LAZY)
  @JoinColumn(name = "employe_id")
  var employe: Employe? = null

  @ManyToOne(fetch = FetchType.LAZY)
  @JoinColumn(name = "caisse_id")
  var caisse: Caisse? = null

  @Column(name = "date_retour")
  var dateRetour: LocalDateTime? = null

  @ColumnDefault("0")
  @Column(name = "supprimer")
  var supprimer: Int? = 0
}

File: backend_py/pharmaPython/app/models/retour_produit.py
from sqlalchemy import Column, Integer, ForeignKey, DateTime
from sqlalchemy.orm import relationship
from app.core.db import Base

class RetourProduit(Base):
  __tablename__ = "retour_produit"

  id = Column(Integer, primary_key=True, index=True)

  vente_id = Column(Integer, ForeignKey("vente.id"))
  vente = relationship("Vente")
  # vente = relationship("Vente", back_populates="retours_produit")

  employe_id = Column(Integer, ForeignKey("employe.id"))
  employe = relationship("Employe")
  # employe = relationship("Employe", back_populates="retours_produit")

  caisse_id = Column(Integer, ForeignKey("caisse.id"))
  caisse = relationship("Caisse")

  date_retour = Column(DateTime, nullable=True)
  supprimer = Column(Integer, default=0)
